File: libgomp/testsuite/libgomp.fortran/task-reduction-17.f90
! { dg-do run }

module m
  integer a
end module m

program main
  use omp_lib
  use m
  implicit none

  !$omp task final (.true.)
    if (.not. omp_in_final ()) &
      error stop
    !$omp task
      if (.not. omp_in_final ()) &
        error stop
      !$omp taskgroup task_reduction (+: a)
        if (.not. omp_in_final ()) &
          error stop
        !$omp task in_reduction (+: a)
          a = a + 1
          if (.not. omp_in_final ()) &
            error stop
        !$omp end task
      !$omp end taskgroup
      if (.not. omp_in_final ()) &
        error stop
      !$omp taskwait
    !$omp end task
  !$omp end task
end

File: libgomp/testsuite/libgomp.c-c++-common/task-reduction-17.c
/* { dg-do run } */

#include <omp.h>
#include <stdlib.h>

int a;

int
main ()
{
  #pragma omp task final (1)
  {
    if (!omp_in_final ())
      abort ();
    #pragma omp task
    {
      if (!omp_in_final ())
	abort ();
      #pragma omp taskgroup task_reduction (+: a)
      {
	if (!omp_in_final ())
	  abort ();
	#pragma omp task in_reduction (+: a)
	{
	  ++a;
	  if (!omp_in_final ())
	    abort ();
	}
      }
      if (!omp_in_final ())
	abort ();
      #pragma omp taskwait
    }
  }
  return 0;
}
